prow/cmd/generic-autobumper/bumper/bumper.go

```diff
@@ -786,44 +786,81 @@ func GitCommitAndPush(remote, remoteBranch, name, email, message string, stdout,
 func GitCommitSignoffAndPush(remote, remoteBranch, name, email, message string, stdout, stderr io.Writer, signoff bool) error {
 	logrus.Info("Making git commit...")
 
-	if err := Call(stdout, stderr, gitCmd, "add", "-A"); err != nil {
-		return fmt.Errorf("failed to git add: %w", err)
-	}
-	commitArgs := []string{"commit", "-m", message}
-	if name != "" && email != "" {
-		commitArgs = append(commitArgs, "--author", fmt.Sprintf("%s <%s>", name, email))
-	}
-	if signoff {
-		commitArgs = append(commitArgs, "--signoff")
-	}
-	if err := Call(stdout, stderr, gitCmd, commitArgs...); err != nil {
-		return fmt.Errorf("failed to git commit: %w", err)
-	}
 	if err := Call(stdout, stderr, gitCmd, "remote", "add", forkRemoteName, remote); err != nil {
 		return fmt.Errorf("failed to add remote: %w", err)
 	}
+	// Stash changes and create a branch the commit will live
+	if err := Call(stdout, stderr, gitCmd, "stash"); err != nil {
+		return fmt.Errorf("failed to git stash: %w", err)
+	}
+	if err := Call(stdout, stderr, gitCmd, "checkout", "-b", remoteBranch); err != nil {
+		return fmt.Errorf("failed to git stash: %w", err)
+	}
+	// Checkout most recent bump PR
 	fetchStderr := &bytes.Buffer{}
-	var remoteTreeRef string
+	// Construct rebasePR branch name from remote Branch so they will never be the same
+	rebasedPR := remoteBranch + "REBASE"
+	needToDiff := true
 	if err := Call(stdout, fetchStderr, gitCmd, "fetch", forkRemoteName, remoteBranch); err != nil {
 		logrus.Info("fetchStderr is : ", fetchStderr.String())
 		if !strings.Contains(strings.ToLower(fetchStderr.String()), fmt.Sprintf("couldn't find remote ref %s", remoteBranch)) {
 			return fmt.Errorf("failed to fetch from fork: %w", err)
+		} else {
+			needToDiff = false
 		}
 	} else {
-		var err error
-		remoteTreeRef, err = getTreeRef(stderr, fmt.Sprintf("refs/remotes/%s/%s", forkRemoteName, remoteBranch))
```

Comment:
The precise reason of the whole fetching remote tree ref and comparing it with the local one is to not push no-op changes. What exactly is the problem with that you are seeing?